src/router_core/router_core_thread.c

```diff
@@ -123,6 +123,33 @@ void qdr_modules_finalize(qdr_core_t *core)
 }
 
 
+/*
+ * router_core_process_background_action_LH
+ *
+ * Process up to one available background action.
+ * Return true iff an action was processed.
+ */
+static bool router_core_process_background_action_LH(qdr_core_t *core)
+{
+    qdr_action_t *action = DEQ_HEAD(core->action_list_background);
+
+    if (!!action) {
+        DEQ_REMOVE_HEAD(core->action_list_background);
+        if (action->label)
```

Comment:
Can the call to qd_log be moved to the unlocked section?  qd_log() takes locks and IIRC does sync I/O.

Reply:
Yes, thanks.  I'll change that.